src/actions/MapActions.js

```diff
@@ -65,10 +66,19 @@ export const updateRegion = (newRegion) => {
 };
 
 export const getSiteDetail = ({selectedSite, navigate}) => {
-    navigate(navKeys.SITE_DETAIL);
+    if (navigate) {
+        navigate(navKeys.SITE_DETAIL);
+    }
 
-    return {
-        type: SELECTED_SITE_UPDATE,
-        payload: {selectedSite}
+    if(selectedSite){
+        return {
+            type: SELECTED_SITE_UPDATE,
+            payload: {selectedSite}
+        }
+    } else {
+        return {
+            type: SELECTED_SITE_CLEARED
+        }
```

Comment:
This action is now used in several places with varying use of the arguments. If we're passed a `navigate` object, it means that we want to redirect to the detail screen. If not, then we either want to just set a `selectedSite` (If we click on a Marker on the MapView), or if we also haven't been passed a `selectedSite` then the user clicked on the X on the site preview and wants to unselect the `selectedSite`.